x/poolmanager/create_pool.go

```diff
@@ -122,18 +119,18 @@ func (k Keeper) createPoolZeroLiquidityNoCreationFee(ctx sdk.Context, msg types.
 		return nil, err
 	}
 
-	// Create and save the pool's module account to the account keeper.
-	// This utilizes the pool address already created and validated in the previous steps.
-	if err := osmoutils.CreateModuleAccount(ctx, k.accountKeeper, pool.GetAddress()); err != nil {
-		return nil, fmt.Errorf("creating pool module account for id %d: %w", poolId, err)
-	}
```

Comment:
Note to reviewer: this is moved to be after `InitializePool` because x/cosmwasmpool address is set in `InitializePool`

I confirmed that there is no dependency in `x/gamm` or `x/concentrated-liquidity` on having this module created before `InitializePool` is called.